server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java

```diff
@@ -107,16 +110,27 @@
     textAccumulator.push(new StringBuilder());
     this.messageService = messageService;
     this.recursiveReplaceStmtStack = recursiveReplaceStmtStack;
+    this.replacingClauses = replacingClauses;
+  }
+
+  @Override
+  public Deque<StringBuilder> getTextAccumulator() {
+    return textAccumulator;
   }
 
   @NonNull
   @Override
   public ExtendedDocument getResult() {
+    if (!replacingClauses.isEmpty()) {
```

Comment:
StringUtils.isNotEmpty(replacingClauses)  will also prevent NPE

Reply:
replacingClauses is a List. I don't think StringUtils would be good, code readability-wise.

Reply:
> StringUtils

Yes, I mean (in my mind) CollectionUtils, but that library of Apache we do not have ;)